.github/workflows/lean4checker.yml

```diff
@@ -78,6 +78,15 @@ jobs:
           cd ..
           lake env lean4checker/.lake/build/bin/lean4checker
 
+      - id: ci_url
+        run: |
+          url=https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}
+          job_no="$(curl -s "${url}" |
+            # extract the CI run number from the unique line that matches `  href=".../job/[job number]"`
+            awk -v url="${url}" -F'/' '/^ *href/ {gsub(/"/, "", $NF); printf("summary<<EOF\n%s/job/%s\nEOF", url, $NF)}')"
+          printf '%s\n' "${job_no}"
+          printf '%s\n' "${job_no}" >> "$GITHUB_OUTPUT"
```

Comment:
Entirely untested, but GPT says this could be:
```
- id: ci_url
  run: |
    url="https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"
    job_no="${{ github.job }}"
    full_url="${url}/jobs/${job_no}"
    echo "job_url=${full_url}" >> "$GITHUB_ENV"
 ```
 which is much nicer if true.

Reply:
Sadly, `github.job` is the name of the job, not the number.  I had googled a bit, but could not find a GitHub API name for the job number.